db/version_set.cc

```diff
@@ -2369,7 +2371,7 @@ Status VersionSet::LogAndApply(ColumnFamilyData* column_family_data,
 
     TEST_SYNC_POINT("VersionSet::LogAndApply:WriteManifest");
     if (!w.edit_list.front()->IsColumnFamilyManipulation() &&
-        db_options_->max_open_files == -1) {
+        mutable_db_options_->max_open_files == -1) {
```

Comment:
We need to obtain the value of mutable_db_options_->max_open_files before unlock db mutex.